test/builder/QuarkBuilderBorrow.t.sol

```diff
@@ -0,0 +1,877 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.23;
+
+import "forge-std/Test.sol";
+import "forge-std/console.sol";
+
+import {QuarkBuilderTest, Accounts, PaymentInfo, QuarkBuilder} from "test/builder/lib/QuarkBuilderTest.sol";
+
+import {Actions} from "src/builder/Actions.sol";
+import {CCTPBridgeActions} from "src/BridgeScripts.sol";
+import {CodeJarHelper} from "src/builder/CodeJarHelper.sol";
+import {CometSupplyMultipleAssetsAndBorrow} from "src/DeFiScripts.sol";
+import {Paycall} from "src/Paycall.sol";
+import {Strings} from "src/builder/Strings.sol";
+
+contract QuarkBuilderBorrowTest is Test, QuarkBuilderTest {
+    uint256 constant BLOCK_TIMESTAMP = 123_456_789;
+    address constant COMET = address(0xc3);
+
+    function borrowIntent_(
+        uint256 amount,
+        string memory assetSymbol,
+        uint256 chainId,
+        uint256[] memory collateralAmounts,
+        string[] memory collateralTokenSymbols
+    ) internal pure returns (QuarkBuilder.BorrowIntent memory) {
+        return QuarkBuilder.BorrowIntent({
+            amount: amount,
+            assetSymbol: assetSymbol,
+            blockTimestamp: BLOCK_TIMESTAMP,
+            borrower: address(0xa11ce),
+            chainId: chainId,
+            collateralAmounts: collateralAmounts,
+            collateralTokenSymbols: collateralTokenSymbols,
+            comet: COMET
+        });
+    }
+
+    function stringArray(string memory string0, string memory string1, string memory string2, string memory string3)
```

Comment:
these functions are a little hacky, I suppose, but it made it nice to be able to write:

```
            assetSymbols: stringArray("USDC", "USDT", "LINK", "WETH"),
            assetBalances: uintArray(0, 0, 10e18, 0)
```